### PR TITLE
Route Timers, Block Etherwarps, Move waypoint type, New nametag style

### PR DESCRIPTION
**Pull request by SubAt0m1c**
Fixed issues where teammate entities werent grabbed, and update their position when they enter the world again instantly.

### CODE REVIEW

**Comment by Queen** [comment]
downvote this pr

File: odinmain/src/main/kotlin/me/odinmain/commands/impl/DungeonWaypointCommand.kt [hunk 1]
```diff
@@ -40,6 +40,13 @@ val dungeonWaypointsCommand = commodore("dwp", "dungeonwaypoints") {
         } ?: modMessage("Invalid type!")
     }
 
+    literal("timer").runs { type: String ->
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/commands/impl/DungeonWaypointCommand.kt:43]: add color codes

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/dungeonwaypoints/SecretWaypoints.kt [hunk 2]
```diff
@@ -49,8 +56,10 @@ object SecretWaypoints {
         val pos = Vec3(packet.x, packet.y, packet.z)
         if (pos.distanceTo(etherpos) > 3) return
         val room = DungeonUtils.currentFullRoom ?: return
-        val vec = etherpos.subtractVec(x = room.clayPos.x, z = room.clayPos.z).rotateToNorth(room.room.rotation)
-        getWaypoints(room).find { wp -> wp.toVec3().equal(vec) && wp.type == WaypointType.ETHERWARP }?.let {
+        val vec = room.getRelativeCoords(etherpos)
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/dungeonwaypoints/SecretWaypoints.kt:59]: don't define variables if they are only used once

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/dungeonwaypoints/SecretWaypoints.kt [hunk 3]
```diff
@@ -61,15 +70,17 @@ object SecretWaypoints {
 
     private fun clickSecret(pos: Vec3, distance: Int, block: IBlockState? = null) {
         val room = DungeonUtils.currentFullRoom ?: return
-        val vec = pos.subtractVec(x = room.clayPos.x, z = room.clayPos.z).rotateToNorth(room.room.rotation)
+        val vec = room.getRelativeCoords(pos)
 
+        val waypoints = getWaypoints(room)
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/dungeonwaypoints/SecretWaypoints.kt:75]: why define it if its only used once

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/dungeonwaypoints/SecretWaypoints.kt [hunk 4]
```diff
@@ -86,4 +97,40 @@ object SecretWaypoints {
...
+        waypoints.find { wp -> wp.toVec3().addVec(y = 0.5).distanceTo(vec) <= 2 && wp.type == WaypointType.MOVE && !wp.clicked }?.let { wp ->
+            wp.timer?.let { if (handleTimer(wp, waypoints, room)) wp.clicked = true else return } ?: run { wp.clicked = true }
+            setWaypoints(room)
+            devMessage("clicked ${wp.toVec3()}")
+            glList = -1
+        }
+    }
+
+    private fun handleTimer(waypoint: DungeonWaypoint, waypoints: MutableList<DungeonWaypoint>, room: FullRoom): Boolean {
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/dungeonwaypoints/SecretWaypoints.kt:114]: add color codes

**Review by odtheking — COMMENTED** [review]
half good review make sure to just not define variables if they are only used once

**Comment by SubAt0m1c** [comment]
The waypoints value is reused in handleTimer to clear clicked checkpoints.

File: odinmain/src/main/kotlin/me/odinmain/utils/render/RenderUtils.kt [hunk 5]
```diff
@@ -350,7 +350,7 @@ object RenderUtils {
             GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA)
 
             for (box in boxes) {
-                if (box.clicked) continue
+                if (box.clicked || box.color.alpha <= 0f) continue
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/utils/render/RenderUtils.kt:353]: use isTransparent

**Pull request merged**